tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,419 @@
+        image_spec (str): The image specification for testing.
+    """
+    log.info(
+        f"Test to remove ceph {service_obj.daemon_type} service along with pwl cache test"
+    )
+
+    def remove_service():
+        cmd = f"ceph orch host label rm {host} {service_obj.daemon_type}"
+        cache_client.exec_command(cmd=cmd, output=True)
+
+        if not service_obj.remove_mon_service(host=host):
+            log.error(f"Could not remove the {service_obj.daemon_type} host")
+            return 1
+        log.info(f"{service_obj.daemon_type} removed successfully")
+
+    with parallel() as p:
+        p.spawn(remove_ceph_service)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def add_ceph_service(service_obj, host, cache_client, pwl, config, image_spec):
+    """
+    Add a Ceph service (e.g., Mon) from a host and verify the
+    persistent write cache in parallel.
+
+    Args:
+        service_obj: The Ceph service object (e.g., Mon, OSD).
+        host (str): The name of the host where the service will be removed.
+        cache_client (Node): The client node for cache testing.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration. Default is None.
+        image_spec (str): The image specification for testing.
+    """
+    log.info(
+        f"Test to add ceph {service_obj.daemon_type} service along with pwl cache test"
+    )
+
+    def add_service():
+        cmd = f"ceph orch host label add {host} {service_obj.daemon_type}"
+        cache_client.exec_command(cmd=cmd, sudo=True)
+        time.sleep(10)
+
+        if not service_obj.check_mon_exists_on_host(host=host):
+            log.error(f"Could not add {service_obj.daemon_type} service")
+            return 1
+        log.info(f"Added {service_obj.daemon_type} service successfully")
+
+    with parallel() as p:
+        p.spawn(add_service)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def remove_osd_service_with_cache_test(
+    rados_obj, ceph_cluster, pwl, config, cache_client, image_spec
+):
+    """
+    Test to remove OSD service along with verifying the persistent write cache.
+
+    Args:
+        rados_obj: RadosOrchestrator object for interacting with the Ceph Orchestrator.
+        ceph_cluster: Ceph cluster object.
+        pwl: PersistentWriteAheadLog instance.
+        config: Test configuration.
+        cache_client: Client node for cache testing.
+        image_spec: Image specification for testing.
+    """
+    log.info("Test to remove OSD service along with pwl cache test")
+
+    pool_name = config["rep_pool_config"]["pool"]
+    pg_set = rados_obj.get_pg_acting_set(pool_name=pool_name)
+    target_osd = pg_set[0]
+    host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=target_osd)
+
+    with parallel() as p:
+        p.spawn(utils.osd_remove, ceph_cluster, target_osd)
```

Comment:
This implementation still doesn't catch any return value and throw exception right? As we discussed.